### ISSUE TITLE
Head state missing, repairing chain

### ISSUE THREAD

**Issue by rajatkeshar2** [Consensys-Incorporated/quorum#843]
Hi there,

I have created an Clique POA consortium on the 3 node. I am getting stuck at some random interval (5-6 hours) one of the node gets down.

**Curent Cluster Size:** 3 Nodes
**Current Numbers of Blocks in Cluster:** 289080

#### System information

Geth version: 1.8.18-stable
Git Commit: 26d381511ed249a919d87ed3de1f305e278ac29b
Quorum Version: 2.2.5
Architecture: amd64
Protocol Versions: [63 62]
Network Id: 1337
Go Version: go1.9.7
Operating System: linux

#### Expected behaviour

Node should be down at run time.

#### Actual behaviour

One of the node gets down at some random interval. 

Sometimes I am getting this logs **"Head state missing, repairing chain"**

![image](https://user-images.githubusercontent.com/46245745/66632676-a30d9700-ec26-11e9-86a7-beb54ba91b2d.png)

And sometimes I am getting this logs, 

![image](https://user-images.githubusercontent.com/46245745/66632729-bc164800-ec26-11e9-8858-b446cfd82fe7.png)

#### Steps to reproduce the behaviour

1. Create a consortium of 3 nodes.
2. Continuously push transactions into the network with the rate 10-20 TPS.

Please let me know how can we resolve this Issue.

**Comment by prd-fox** [comment]
Hi @rajatkeshar2 

Does the crash happen first, and then you see the log statement upon restarting the node?

Version `2.3.0` was released last week that contains fixes for the crash you described above, would you be able to upgrade and see if that helps your problem?

As for the `Head state missing, repairing chain`, this is when the state hasn't been flushed to disk properly (such as after a crash) and it needs to go back and replay the blocks from the last time is did have the full state available. You should find that it will just need to sync for a bit and then operate as normal - let me know if this isn't the case.

**Comment by rajatkeshar2** [comment]
Hi @prd-fox Thanks for the reply.

Both Issues are coming at random time interval. After restarting the node everything is working fine but at some time period any one of the issue is happning.

I have also tried with `v2.2.3` there were crash is happning after that I have moved to the `v2.2.5` 

These are the options I am using for starting nodes:

ARGS="--nodiscover --verbosity 6 --syncmode full --gcmode archive --networkid $NETWORK_ID --mine --minerthreads 1 --rpc --rpcaddr 0.0.0.0 --rpcapi admin,db,eth,debug,miner,net,shh,txpool,personal,web3,quorum,clique"

PRIVATE_CONFIG=node/tesseraConfig/tm.ipc nohup geth --datadir node $ARGS --rpcport 24000 --port 24001 --unlock 0 --password password.txt 2>>logs/${node}.log &

**Comment by prd-fox** [comment]
Affected versions are `2.2.2` up to and including `2.2.5`. Version `2.3.0` is the version where the crash is fixed, so you will need to use that version to confirm.



> Both Issues are coming at random time interval.

Is the `Head state missing, repairing chain` really coming at random time intervals? It should only appear at startup, and only after a non-clean shutdown.

**Comment by prd-fox** [comment]
Hi @rajatkeshar2 

Not heard anything on this issue for some time, so closing it. If you have tried the new version and still get these issues, please reopen or create a new issue.

Thanks.
